numpy/ma/core.py

```diff
@@ -3802,7 +3802,10 @@ def __iadd__(self, other):
         else:
             if m is not nomask:
                 self._mask += m
-        ndarray.__iadd__(self._data, np.where(self._mask, 0, getdata(other)))
+        ndarray.__iadd__(
```

Comment:
For that matter, `self._data[Ellipsis] += ...` works, as well. Of course, we can replace `Ellipsis` with `...`, I didn't do so here to avoid unnecessary confusion.